packages/host/app/resources/auto-attached-card.ts

```diff
@@ -1,40 +1,93 @@
-import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
 
 import { Resource } from 'ember-resources';
 
+import { TrackedSet } from 'tracked-built-ins';
+
+import type { StackItem } from '@cardstack/host/lib/stack-item';
+
 import { type CardDef } from 'https://cardstack.com/base/card-api';
 
 interface Args {
   named: {
-    lastTopMostCard: CardDef | undefined; // top-most card from the card stack
+    stackItems: StackItem[];
     attachedCards: CardDef[] | undefined; // cards manually attached in ai panel
   };
 }
 
 /**
- * Manages the auto-attachment of the top-most card in consideration of user-actions of manually
+ * Manages the auto-attachment of cards within our stack consideration of user-actions of manually
  * removing and attaching new cards in the ai panel
  */
+// when removing manually added, it will be removed and not still auto-attached
+// when drilling down, it will be auto attached (unless already manually added)
+// removing auto-attached removes it. unless its opened in another stack
+// manually adding a card will thats open or a stack will mean its no longer auto-attached
+// opening two cards in stack will auto-attach both
+// clearing multiple cards work too
 
 export class AutoAttachment extends Resource<Args> {
-  @tracked card: CardDef | undefined;
-  private lastCard: CardDef | undefined;
+  cards: TrackedSet<CardDef> = new TrackedSet(); // auto-attached cards
+  private lastStackedItems: StackItem[] = [];
+  private lastRemovedCards: Set<string> = new Set(); // internal state, changed from the outside. It tracks, everytime a card is removed in the ai-panel
 
   modify(_positional: never[], named: Args['named']) {
-    const { lastTopMostCard, attachedCards } = named;
-    if (lastTopMostCard === undefined) {
-      this.card = undefined;
-      this.lastCard = undefined;
-      return;
+    const { stackItems, attachedCards } = named;
+    if (this.stackItemsChanged(stackItems)) {
+      // we must be sure to clear the lastRemovedCards state so cards can be auto-attached again
```

Comment:
This is what I mean here. It's not great but I don't have a good solution to deal with this currently 

https://github.com/cardstack/boxel/assets/8165111/b346f7a3-baa3-4711-bf75-63e7dfe4dda4